### PR TITLE
adds shared_ptr overloads for mjbots constructors

### PR DESCRIPTION
**Pull request by jdcaporale**
(Pull Request 1 of 3 in a sequence)

-  Adds shared_pointer vector constructors to the mjbots classes (robot and control loop).
-  Adds make_share_vector helper function to help with construction.

### CODE REVIEW

**Review by ethanmusser — APPROVED** [review]
LGTM.

File: src/mjbots_robot_interface.cpp [hunk 1]
```diff
@@ -96,10 +101,8 @@ MjbotsRobotInterface::MjbotsRobotInterface(const std::vector<JointMoteus> &joint
   moteus_data_.commands = {commands_.data(), commands_.size()};
   moteus_data_.replies = {replies_.data(), replies_.size()};
   moteus_data_.timeout = timeout_;
-
 }
 
-
```
Comment (ShaneRozenLevy) [inline src/mjbots_robot_interface.cpp:102]: nit: white space

File: src/mjbots_robot_interface.cpp [hunk 2]
```diff
@@ -96,10 +101,8 @@ MjbotsRobotInterface::MjbotsRobotInterface(const std::vector<JointMoteus> &joint
   moteus_data_.commands = {commands_.data(), commands_.size()};
   moteus_data_.replies = {replies_.data(), replies_.size()};
   moteus_data_.timeout = timeout_;
-
```
Comment (ShaneRozenLevy) [inline src/mjbots_robot_interface.cpp:99]: nit white space

**Review by ShaneRozenLevy — APPROVED** [review]
LGTM, some very minor nits that makes it sad we don't have an auto formatter.

File: src/mjbots_robot_interface.cpp (hunk 1 above)
Comment (ethanmusser) [reply src/mjbots_robot_interface.cpp:102]: Agreed, we should do a formatting run (see #35).

Comment (jdcaporale) [reply src/mjbots_robot_interface.cpp:102]: Yea Ethan has been pushing for more auto formatting and I've been slowly coming around to the idea.

**Pull request merged**